examples/mlmc.cpp

```diff
@@ -275,18 +278,12 @@ int main(int argc, char* argv[])
     rhs_fine.GetBlock(0) = 0.0;
     rhs_fine.GetBlock(1) = rhs_u_fine;
 
-    const int num_fine_vertices = vertex_edge.Height();
-    const int num_fine_edges = vertex_edge.Width();
-    const int num_aggs = partitioning.Max() + 1; // this can be wrong if there are empty partitions
-    if (myid == 0)
-    {
-        std::cout << "fine graph vertices = " << num_fine_vertices << ", fine graph edges = "
-                  << num_fine_edges << ", coarse aggregates = " << num_aggs << std::endl;
-    }
-    unique_ptr<TwoLevelSampler> sampler;
+    const int num_levels = upscale_param.max_levels;
+    unique_ptr<MultilevelSampler> sampler;
     if (std::string(sampler_type) == "simple")
     {
-        sampler = make_unique<SimpleSampler>(num_fine_vertices, num_aggs);
+        MFEM_ASSERT(num_levels == 2, "SimpleSampler only implemented 2-level here!");
```

Comment:
Seems to me that you have made a multilevel version of `SimpleSampler` that takes a `vector` of sizes as input. The sizes should be available in `GraphTopology`, maybe need a function to put them into a `vector` though. So I think this can be in ML setting without too much effort.

Reply:
I was thinking a function in `Upscale` to collect and output those sizes.